### ISSUE TITLE
Is it possible to extract MBTiles using a POLY file?

### ISSUE THREAD

**Issue by carlos-mg89** [mapbox/node-mbtiles#86]
Hi and thanks for this project. It's extremely helpful and saves tones of time so there's no need to generate MBTiles by regions, and it can be done by higher levels, and then we can cut them to generate smaller MBTiles. To do this, I'm using a bounding box, and the following query:

`tilelive-copy --minzoom=0 --maxzoom=14 --bounds="5.9559,45.818,10.4921,47.8084" planet.mbtiles my-extract-for-switzerland.mbtiles`

This is already super helpful and it is super great. However, I'm wondering if it's possible to use POLY files, instead of a bounding box, to cut the planet.mbtiles (or whatever mbtiles file)?

This would allow creating even smaller MBTiles. At the moment, the only possibility I see to do this is to use the POLY file, with osmosis (for example), to generate a smaller osm.pbf file, and then use that file to create the MBTiles. However, the process to generate MBTiles it's slow and cutting the MBTiles using the above (bounding boxed) is super fast. If it would be possible to use a POLY file instead, it would be just perfect.

Thanks in advance!

**Comment by moodstubos** [comment]
Hi carlos, 
do you have any results of your investigations?
I need exactly the same function :-)

best regards
moods

**Comment by carlos-mg89** [comment]
@moodstubos sadly nothing more than what I wrote here. I could have dedicated some time to figure things out and submit a PR to this project, but I went through the easy and long way.

Use your POLY files to cut your OSM.PBF file and get smaller map files (per state or even smaller areas), then convert it to MBTiles format, and that's it.

Actually, in my current setup, I only cut the files using tilelive-copy using the bounding box, and I assume the output MBTiles would be bigger than they could if the cut was performed using a POLY file. It's not ideal, but it's way faster.
